File: tarumt-nav-app/flutter_app/test/ui/map/models/view_cone_model_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:indoor_navigation/ui/map/models/view_cone_model.dart';

void main() {
  test('creates a symmetric 60-degree fan facing right', () {
    final geometry = createViewConeGeometry(
      fieldOfViewDegrees: 60,
      length: 100,
    );

    expect(
      geometry.path,
      'M 100 100 L 186.603 50 A 100 100 0 0 1 186.603 150 Z',
    );
    expect(geometry.size, 200);
  });

  test('clamps invalid field of view and length inputs', () {
    expect(createViewConeGeometry(fieldOfViewDegrees: 0, length: 0).size, 2);
    expect(
      createViewConeGeometry(fieldOfViewDegrees: 360, length: 20).path,
      startsWith('M 20 20 '),
    );
  });
}

File: tarumt-nav-app/expo-app/src/components/map/actor/viewConeModel.test.ts
import assert from 'node:assert/strict';
import test from 'node:test';

import { createViewConeGeometry } from './viewConeModel';

test('creates a symmetric 60-degree fan facing right', () => {
  assert.deepEqual(
    createViewConeGeometry({ fieldOfViewDegrees: 60, length: 100 }),
    {
      path: 'M 100 100 L 186.603 50 A 100 100 0 0 1 186.603 150 Z',
      size: 200,
    },
  );
});

test('clamps invalid field of view and length inputs', () => {
  assert.equal(
    createViewConeGeometry({ fieldOfViewDegrees: 0, length: 0 }).size,
    2,
  );
  assert.match(
    createViewConeGeometry({ fieldOfViewDegrees: 360, length: 20 }).path,
    /^M 20 20 /,
  );
});
